### PR TITLE
GCF: Add Node 8 background function samples

### PR DESCRIPTION
**Pull request by ericschmidtatwork**
Adding `helloPromise` and `helloSynchronous` samples for Node 8 to illustrate returning a Promise and a synchronously returning a result.

### CODE REVIEW

File: functions/node8/index.js [hunk 1]
```diff
@@ -207,3 +207,37 @@ exports.helloAnalytics = (data, context) => {
...
+// [START functions_background_promise_node8]
+/**
+ * Background Cloud Function that returns a Promise. Note that we don't pass
+ * a "callback" argument to the function.
+ *
+ * @param {object} data The Cloud Functions event data.
+ * @returns {Promise}
+ */
+exports.helloPromise = (data) => {
+  const request = require('request-promise');
```
Comment (ace-n) [inline functions/node8/index.js]: Can you move this outside of the function, and use two region tag blocks? e.g:
```
// [START tag_name]
const library = require('library');
// [END tag_name]

...

// [START tag_name]
exports.useLibrary = () => {
  ...
}
// [END tag_name]

**Comment by ace-n** [comment]
Also - can you add (unit) tests, or should I do that?

**Comment by ericschmidtatwork** [comment]
@ace-n Can you add unit tests? I can't get the tests running locally.

**Comment by ericschmidtatwork** [comment]
Looks good to me, thanks!

**Pull request merged**